### ISSUE TITLE
F# client

### ISSUE THREAD

**Issue by Odonno** [surrealdb/surrealdb.net#7]
* Create a packageable F# library named `SurrealDb.FSharp`
* Should follow the same convention as the C# client (class names, method names, etc...)
* Make use of the F# paradigm (Result type, etc...)
* [Optional] If possible, create a packageable C# library named `SurrealDb.Core` to avoid code duplication
* Create some example projects `SurrealDb.FSharp.Examples.Console` and/or `SurrealDb.FSharp.Examples.WeatherApi`
* [Optional] Add tests and benchmarks projects
* Update documentation
  * Readme file
  * Official documentation website

**Comment by S-E-F** [comment]
Do we need an F# package? Can't you reference any .NET dll from any .NET like language?

**Comment by Odonno** [comment]
> Do we need an F# package? Can't you reference any .NET dll from any .NET like language?

Yes, you can. But the F# implementation could be slightly different than the C# one given the specifity of the language.
It can be seen as a nice-to-have, but for F# developers this issue can be seen as mandatory.

**Comment by S-E-F** [comment]
In that case perhaps we should wait and see how F# developers use the package to gain a better insight into how it would differ from the C# version.

**Comment by vitash** [comment]
There is also a need for the same.
- Writing scripts to run, F# is more extensive than C# ([Scripting with F#](https://learn.microsoft.com/en-us/dotnet/fsharp/tools/fsharp-interactive/#scripting-with-f) `dotnet fsi Script.fsx`), without having to create a new project.
- F# uses `null` less and `Option<T>` more often.
- F#, like Rust, has more types than C#.
``` F#
type Enum1 = A | B | C
type TaggedUnion1 = 
     | Num1 of int
     | Str1 of string
type Record1 = { E1: Enum1; Data: TaggedUnion1 }

let a1 = { Record1.E1 = A; Data = Num1 1111 }
let b1 = { Record1.E1 = B; Data = Str1 "sss" }
```

**Comment by Odonno** [comment]
I don't think anyone is working on it at the moment. Feel free to contribute. I can provide some guidance if needed.
